src/lib/import_export/binary/binary_writer.cpp

```diff
@@ -138,16 +138,28 @@ void BinaryWriter::_write_chunk(const Table& table, std::ofstream& ofstream, con
 
   // Iterating over all segments of this chunk and exporting them
   for (ColumnID column_id{0}; column_id < chunk->column_count(); column_id++) {
-    resolve_data_and_segment_type(
-        *chunk->get_segment(column_id),
-        [&](const auto data_type_t, const auto& resolved_segment) { _write_segment(resolved_segment, ofstream); });
+    resolve_data_and_segment_type(*chunk->get_segment(column_id),
+                                  [&](const auto data_type_t, const auto& resolved_segment) {
+                                    using ColumnDataType = typename decltype(data_type_t)::type;
+                                    using SegmentType = std::decay_t<decltype(resolved_segment)>;
+                                    if constexpr (std::is_same_v<SegmentType, ValueSegment<ColumnDataType>>) {
+                                      _write_segment(resolved_segment, table.column_is_nullable(column_id), ofstream);
+                                    } else {
+                                      _write_segment(resolved_segment, ofstream);
+                                    }
```

Comment:
i now added the column_is_nullable to all the write_segment calls so that we don't need a distinction between segments here. i didn't liked having the column_is_nullable for other segments than value segment because it is not used for the export. but since this part is not performance relevant, i think we can go for simplicity here.